### PR TITLE
Fetch project contacts schema from the database

### PR DESCRIPTION
**Pull request by BCerki**
card: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1359

Found an existing bug while working on this: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1496

### CODE REVIEW

File: app/tests/unit/pages/project-revision/[projectRevision]/index.test.tsx [hunk 1]
```diff
@@ -158,7 +159,7 @@ describe("The Create Project page", () => {
     pageTestingHelper.loadQuery({
       Form() {
         return {
-          jsonSchema: reportingRequirementProdSchema,
+          jsonSchema: projectContactProdSchema,
```
Comment (BCerki) [inline app/tests/unit/pages/project-revision/[projectRevision]/index.test.tsx:162]: I ended up not needing to delve into mock resolver context here, but I added a link to the docs for it in the other db-fetching cards

File: app/components/Form/ProjectContactFormSummary.tsx [hunk 2]
```diff
@@ -117,7 +118,7 @@ const ProjectContactFormSummary: React.FC<Props> = ({
             tagName={"dl"}
             fields={renderDiff ? customFields : fields}
             theme={readOnlyTheme}
-            schema={projectContactSchema as JSONSchema7}
+            schema={node.formByJsonSchemaName.jsonSchema.schema}
```
Comment (tmastrom) [inline app/components/Form/ProjectContactFormSummary.tsx:121]: Do we still want to cast this to `JSONSchema7`

File: app/components/Form/ProjectContactFormSummary.tsx [hunk 3]
```diff
@@ -185,7 +191,9 @@ const ProjectContactFormSummary: React.FC<Props> = ({
               tagName={"dl"}
               theme={readOnlyTheme}
               fields={renderDiff ? customFields : fields}
-              schema={projectContactSchema as JSONSchema7}
+              schema={
+                primaryContact?.node.formByJsonSchemaName.jsonSchema.schema
+              }
```
Comment (tmastrom) [inline app/components/Form/ProjectContactFormSummary.tsx:196]: cast to `JSONSchema7`?

File: app/components/Form/ProjectContactForm.tsx [hunk 4]
```diff
@@ -118,14 +100,44 @@ const ProjectContactForm: React.FC<Props> = (props) => {
...
+          title: node.fullName,
+          enum: [node.rowId],
+          value: node.rowId,
+        } as JSONSchema7Definition;
+      }),
+    };
+    return schema as JSONSchema7;
+  }, [allContacts, contactFormBySlug, areContactsEmpty]);
+
+  console.log("contactSchema", contactSchema);
```
Comment (tmastrom) [inline app/components/Form/ProjectContactForm.tsx:140]: probably left over from development

File: app/components/Form/ProjectContactFormSummary.tsx (hunk 3 above)
Comment (BCerki) [reply app/components/Form/ProjectContactFormSummary.tsx:196]: I think the db schemas are already type `JSONSchema7`

File: app/components/Form/ProjectContactForm.tsx (hunk 4 above)
Comment (BCerki) [inline app/components/Form/ProjectContactForm.tsx:140]:
```suggestion
```

Comment (BCerki) [inline app/components/Form/ProjectContactForm.tsx:140]:
```suggestion
```

**Review by tmastrom — DISMISSED** [review]

**Review by tmastrom — APPROVED** [review]

**Pull request merged**